Adds link to scanner on badge

ChaosExAnima opened the pull request:
#### Changes proposed in this Pull Request

* Adds a link to the scanner when the badge is showing.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Load a site with Scan in Cloud.
* On Backup page with the Scan menu closed, create a threat or progress meter.

  Example action:
  ```js
  {
    type: 'JETPACK_SCAN_UPDATE',
    siteId: 12345,
    payload: {
      state: 'idle',
      threats: [{ id: 123, signature: 'TEST', description: 'Test Threat', status: 'current', firstDetected: '2020-01-01', fixable: false }],
      credentials: [],
      mostRecent: {
        timestamp: '2020-05-11T20:08:10+00:00',
        duration: 220,
        progress: 50,
        error: false,
        isInitial: false
      }
    }
  }
  ```
* Click on the badge.

Fixes 1151678672052943-as-1174867022546273.

ChaosExAnima commented:
@rcanepa - Can you retest? I fixed an incorrect selector in 3a7a229.

enejb commented:
Why place not wrap the <Badge> by link instead of inserting the link into the badge? 

The way it currently works is that the only the text actually will take you to the Scanner page. 
If you miss slightly that and click on the background you will just expand the Navigation.

ChaosExAnima commented:
@enejb - Not a bad suggestion, I can look into that.

ChaosExAnima commented:
Addressed feedback from @enejb. The clickable area does not cover the entire menu header as that would involve using an entirely new component, but it does cover the whitespace and header text.

enejb commented:
This worked beautifully for me! Nicely done!
